src/cinder/gl/GlslProg.cpp

```diff
@@ -697,10 +718,20 @@ void GlslProg::loadShader( const string &shaderSource, const fs::path &shaderPat
 			}
 			log += "\n" + ss.str();
 		}
-#endif		
+#endif
+		// Since the GlslProg destructor will not be called after we throw, we must delete all
+		// owned GL objects here to avoid leaking. Any other attached shaders will be cleaned up
+		// when the program is deleted
+		glDeleteShader( handle );
+		glDeleteProgram( mHandle );
+
 		throw GlslProgCompileExc( log, shaderType );
 	}
+
 	glAttachShader( mHandle, handle );
+	glDeleteShader( handle );
```

Comment:
Ah. Actually there's a problem with waiting to delete the shaders. If a subsequent shader fails to compile or linking the program fails they will not be deleted along with the program. Deleting them ahead of time scopes their lifetime to the program.